src/views/Entities/ExternalServices.vue

```diff
@@ -0,0 +1,381 @@
+          { label: 'TLS', key: 'tlsEnabled' }
+        ],
+        data: []
+      },
+      tabs: [
+        {
+          hash: '#overview',
+          title: 'Overview'
+        },
+        {
+          hash: '#yaml',
+          title: 'YAML'
+        }
+      ],
+      entity: [],
+      rawEntity: null,
+      firstEntity: null,
+      pageSize: this.$pageSize,
+      pageOffset: null,
+      next: null,
+      hasNext: false,
+      previous: []
+    }
+  },
+  computed: {
+    tabGroupTitle () {
+      const entity = this.entity
+
+      if (entity) {
+        return `External Service: ${entity.name}`
+      } else {
+        return null
+      }
+    },
+    entityOverviewTitle () {
+      const entity = this.entity
+
+      if (entity) {
+        return `Entity Overview for ${entity.name}`
+      } else {
+        return null
+      }
+    },
+    formattedRawEntity () {
+      const entity = this.formatForCLI(this.rawEntity)
+
+      return entity
+    },
+    shareUrl () {
+      const urlRoot = `${window.location.origin}#`
+      const entity = this.entity
+
+      const shareUrl = () => {
+        if (this.$route.query.ns) {
+          return this.$route.fullPath
+        }
+
+        return `${urlRoot}${this.$route.fullPath}?ns=${entity.name}`
+      }
+
+      return shareUrl()
+    }
+  },
+  watch: {
+    '$route' (to, from) {
+      this.init()
+    }
+  },
+  beforeMount () {
+    this.init()
+  },
+  methods: {
+    init () {
+      this.loadData()
+    },
+    goToPreviousPage () {
+      this.pageOffset = this.previous.pop()
+      this.next = null
+
+      this.loadData()
+    },
+    goToNextPage () {
+      this.previous.push(this.pageOffset)
+      this.pageOffset = this.next
+      this.next = null
+
+      this.loadData()
+    },
+    tableAction (ev) {
+      const data = ev
+
+      // reset back to the first tab
+      this.$store.dispatch('updateSelectedTab', this.tabs[0].hash)
+
+      // set the active table row
+      this.$store.dispatch('updateSelectedTableRow', data.name)
+
+      // load the data into the tabs
+      this.getEntity(data)
+    },
+    loadData () {
+      this.isLoading = true
+
+      const mesh = this.$route.params.mesh || null
+      const query = this.$route.query.ns || null
+
+      const params = {
+        size: this.pageSize,
+        offset: this.pageOffset
+      }
+
+      const endpoint = () => {
+        if (mesh === 'all') {
+          return this.$api.getAllExternalServices(params)
+        } else if ((query && query.length) && mesh !== 'all') {
+          return this.$api.getExternalService(mesh, query, params)
+        }
+
+        return this.$api.getAllExternalServicesFromMesh(mesh)
+      }
+
+      const getExternalServices = () => {
+        return endpoint()
+          .then(response => {
+            const items = () => {
+              const r = response
+
+              if ('total' in r) {
+                if (r.total !== 0 && r.items && r.items.length > 0) {
```

Comment:
I prefer more explicit expressions as you provided here with `r.items.length > 0` but in other places I see you are using less type strict `r.items.length` which will check for `truthy` values, and if you are sure of the type of `r.items` (if `r.items` is present), you can use this more concise version (`r.items.length`)